### PR TITLE
euclidean_distance_update

### PR DESCRIPTION
**Pull request by Lastrophysicien**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [ ] A clear docstring
  - [x] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [ ] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [ ] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/challenge_34/euclidean_distance.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 01/03/2025
+
```
Comment (MPKenley) [inline solutions/challenge_34/euclidean_distance.py:5]: Docstrings: 


Feedback: It would be useful to put a docstring on the top of the module and

File: solutions/tests/challenge_34/test_euclidean_distance.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class TestEuclideanDistance(unittest.TestCase):
+    """
+    Unit tests for the `euclidean_distance` function.
+    This class includes test cases for 2D, 3D, and error scenarios like
+    different dimensions and negative coordinates.
+    """
+
+    def test_same_point(self):
```
Comment (MPKenley) [inline solutions/tests/challenge_34/test_euclidean_distance.py:22]: Tests names: 

Feedback: the name of the tests could be more clear and match well with the docstrings. How do you think about changing `test_same_point` to `test_with_same_point` ?

File: solutions/tests/challenge_34/test_euclidean_distance.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        """
+        Test case where both points are the same.
+        The Euclidean distance between the same points should be 0.
+        """
+        p1 = (1, 2, 3)
+        p2 = (1, 2, 3)
```
Comment (MPKenley) [inline solutions/tests/challenge_34/test_euclidean_distance.py:28]: No logic in the tests file: 


Feedback: Avoid to put logic in the file. Parameters must define in the program. The test must just be imported from the module.

File: solutions/challenge_34/euclidean_distance.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+def euclidean_distance(point1, point2):
+    """
+    Calculates the Euclidean distance between two points point1
+    and point2 in an n-dimensional space.
+
+    :param point1: List or tuple representing the coordinates of the first point.
+    :param point2: List or tuple representing the coordinates of the second point.
+    :return: The Euclidean distance between point1 and point2.
```
Comment (MPKenley) [inline solutions/challenge_34/euclidean_distance.py:31]: Module:


Feedback: It would be great if you could add three passings doctests at beginning of the python file.

**Review by MPKenley — CHANGES_REQUESTED** [review]
Try your best to pass all the Ruff and pylint tests.

**Review by MPKenley — APPROVED** [review]
